nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/MachineMemoryUsageHealthCheck.java

```diff
@@ -91,110 +91,111 @@ public String getDescription() {
-        else {
-            try {
-                OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
-                Long totalPhysicalMemSize = invokeMethodFor(osBean, "getTotalPhysicalMemorySize");
-                Long freePhysicalMemSize = invokeMethodFor(osBean, "getFreePhysicalMemorySize");
-
-                double usedPercentage = ((double) (totalPhysicalMemSize - freePhysicalMemSize) / totalPhysicalMemSize) *
-                        100;
-
-                result.add(new HealthCheckResultEntry(decideOnStatusWithRatio(usedPercentage),
-                        "Physical Memory Used: " + prettyPrintBytes((totalPhysicalMemSize - freePhysicalMemSize)) + " - " +
-                                "Total Physical Memory: " + prettyPrintBytes(totalPhysicalMemSize) + " - " +
-                                "Memory Used%: " + new DecimalFormat("#.00").format(usedPercentage) + "%"));
-
-            } catch (Exception exception) {
-                result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CHECK_ERROR,
-                        "Operating system methods cannot be invoked for retrieving physical memory usage values",
-                        exception));
-            }
-        }
-
-        return result;
+		return doCheckNonLinux();
     }
 
-    private boolean isLinux() {
+	private HealthCheckResult doCheckLinux() {
+		HealthCheckResult result = new HealthCheckResult();
+		long memTotal = 0;
+		try {
+		    List<String> lines = Files.readAllLines(Paths.get("/proc/meminfo"), StandardCharsets.UTF_8);
+		    if (lines.isEmpty()) {
+		        return result;
+		    }
+		    long memAvailable = 0;
+		    long memFree = 0;
+		    long memActiveFile = 0;
+		    long memInactiveFile = 0;
+		    long memReclaimable= 0;
+		    boolean memAvailableFound = false;
+
+		    for (String line : lines) {
+		        String[] parts = line.split("\\s+");
+
+		        if (parts.length > 1) {
+		            String part = parts[0];
+		            if (MEMAVAILABLE.equals(part)) {
+		                memAvailable = parseMemInfo(parts);
+		                memAvailableFound = true;
+		            }
+		            if (MEMFREE.equals(part)) {
+		                memFree = parseMemInfo(parts);
+		            }
+		            if (MEMTOTAL.equals(part)) {
+		                memTotal = parseMemInfo(parts);
+		            }
+		            if (ACTIVEFILE.equals(part)) {
+		                memActiveFile = parseMemInfo(parts);
+		            }
+		            if (INACTIVEFILE.equals(part)) {
+		                memInactiveFile = parseMemInfo(parts);
+		            }
+		            if (RECLAIMABLE.equals(part)) {
+		                memReclaimable = parseMemInfo(parts);
+		            }
+		        }
+		    }
+
+		    if (!memAvailableFound) {
+		        memAvailable = memFree + memActiveFile + memInactiveFile + memReclaimable;
+		    }
+
+		    double usedPercentage = memTotal == 0L ? 0d :((double) (memTotal - memAvailable) / memTotal) * 100;
```

Comment:
Diff can out a bit unfortunate. I moved the local vars and extracted the main level if-else block into methods on their own. Other than that I did not change the code except for the addition of `memTotal == 0L ? 0d :` here to handle to zero case and avoid div-by-zero.